### PR TITLE
Feature/directory controller

### PR DESCRIPTION
**Pull request by Segelzwerg**
This adds a controller for all routes starting with `/material1/dir`

### CODE REVIEW

**Comment by iTitus** [comment]
please resolve the conflict first

File: src/main/java/mops/presentation/DirectoryController.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+     * @param token      keycloak auth token
+     * @param model      spring view model
+     * @param dirId      id of the parent folder
+     * @param folderName name of the new sub folder
+     * @return object of the folder
+     */
+    @PostMapping("/{dirId}/create")
+    public String createSubFolder(KeycloakAuthenticationToken token,
+                                  Model model,
+                                  @PathVariable("dirId") long dirId,
```
Comment (iTitus) [inline src/main/java/mops/presentation/DirectoryController.java]: same argument as above: maybe rename to `parentDirId`.

File: src/main/java/mops/presentation/DirectoryController.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+     */
+    private final FileService fileService;
+
+    /**
+     * @param token keycloak auth token
+     * @param model spring view model
+     * @param dirId id of the folder
+     * @return route to folder
+     */
+    @GetMapping(path = "/{dirId}")
```
Comment (iTitus) [inline src/main/java/mops/presentation/DirectoryController.java]: `path = ` should not be necessary here. did you try without?

File: src/test/java/mops/presentation/DirectoryControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+     * Necessary bean.
+     */
+    private MockMvc mvc;
+
+    /**
+     * Setups the a Mock MVC Builder.
+     */
+    @BeforeEach
+    public void setUp() {
+        ArrayList<FileInfo> files = new ArrayList<>();
```
Comment (iTitus) [inline src/test/java/mops/presentation/DirectoryControllerTest.java]: please use `List.of()` on both mock configs instead.

File: src/main/java/mops/presentation/DirectoryController.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+     */
+    @PostMapping("/{dirId}/search")
+    public String searchFolder(KeycloakAuthenticationToken token,
+                               Model model,
+                               @PathVariable("dirId") long dirId,
+                               @RequestAttribute("searchQuery") FileQuery query) {
+        final Account account = AccountUtil.getAccountFromToken(token);
+        final List<FileInfo> files = directoryService.searchFolder(account, dirId, query);
+        model.addAttribute("files", files);
+        return "files";
```
Comment (iTitus) [inline src/main/java/mops/presentation/DirectoryController.java]: don't we want this to be a specific template?
so we can show extra info.

File: src/main/java/mops/businesslogic/DirectoryService.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+    /**
+     * Creates a new folder inside a folder.
+     *
+     * @param account user credentials
+     * @param dirId   id of the parent folder
+     * @param dirName name of the new folder
+     * @return id of the new folder
+     */
+    int createFolder(Account account, long dirId, String dirName);
```
[the remark this answers is not in the document]
Comment (Segelzwerg) [reply src/main/java/mops/businesslogic/DirectoryService.java]: yes of course

File: src/main/java/mops/presentation/DirectoryController.java (hunk 2 above)
Comment (Segelzwerg) [reply src/main/java/mops/presentation/DirectoryController.java]: yes it works without it, copy & paste it from documentation

File: src/main/java/mops/presentation/DirectoryController.java (hunk 4 above)
Comment (Segelzwerg) [reply src/main/java/mops/presentation/DirectoryController.java]: I'm not sure, what you meant by that. You want to do a redirect with path variables?

File: src/test/java/mops/presentation/DirectoryControllerTest.java (hunk 3 above)
Comment (Segelzwerg) [reply src/test/java/mops/presentation/DirectoryControllerTest.java]: I don't see the advantage? It is irrelevant what is returned there as long as something is returned.

Comment (iTitus) [reply src/test/java/mops/presentation/DirectoryControllerTest.java]: One should not use `ArrayList<>` as a variable type.
And `List.of()` is the preferred way for empty immutable lists.

File: src/main/java/mops/presentation/DirectoryController.java (hunk 4 above)
Comment (iTitus) [reply src/main/java/mops/presentation/DirectoryController.java]: its fine for now

**Comment by Segelzwerg** [comment]
> `List.of()` and `parentDirId` please ;)

done :trophy:

**Comment by iTitus** [comment]
> 
> 
> > `List.of()` and `parentDirId` please ;)
> 
> done 🏆

no, checkstyle...

File: src/main/java/mops/presentation/DirectoryController.java [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+     * @return object of the folder
+     */
+    @PostMapping("/{parentDirId}/create")
+    public String createSubFolder(KeycloakAuthenticationToken token,
+                                  Model model,
+                                  @PathVariable("parentDirId") long parentDirId,
+                                  @RequestAttribute("folderName") String folderName) {
+        final Account account = AccountUtil.getAccountFromToken(token);
+        final long directoryId = directoryService.createFolder(account, parentDirId
+                , folderName);
```
Comment (iTitus) [inline src/main/java/mops/presentation/DirectoryController.java]:
```suggestion
        final long directoryId = directoryService.createFolder(account, parentDirId, folderName);
```

**Pull request merged**